strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -277,12 +282,12 @@ async def cleanup_operation(self, operation_id: str) -> None:
             return
         generator = self.subscriptions.pop(operation_id)
         task = self.tasks.pop(operation_id)
-        # since python 3.8, generators cannot be reliably closed
-        with suppress(RuntimeError):
-            await generator.aclose()
         task.cancel()
         with suppress(BaseException):
             await task
+        # since python 3.8, generators cannot be reliably closed
+        with suppress(RuntimeError):
+            await generator.aclose()
```

Comment:
```suggestion
            await result_source.aclose()
```

Reply:
Renamed generator in commit: https://github.com/strawberry-graphql/strawberry/pull/2141/commits/0f70ef209bc178def787118d7088ec1b5165a542